docs/pages/blog/making-customizable-components.md

```diff
@@ -0,0 +1,287 @@
+If you’re willing to start from scratch, that's a nice approach.
+Use one hook to manage filtering, another one to manage the sorting, and then build your UI using returned values.
+
+This approach can scale because you can scope parameters to individual features.
+The filtering hooks will only take into account parameters impacting the filtering, and so on—so you can split your code feature by feature.
+
+But because this is a fully custom approach, it will take the most amount of work relative to all other options to construct a functional UI.
+If your main priority is to get up and running quickly, then this may not be a viable solution.
+
+#### Subdivide your components
+
+Another approach I like is to provide subcomponents.
+This is what we do for MUI Core components such as the [Menu](https://mui.com/material-ui/react-menu/).
+
+This is also the approach used by [react-admin](https://marmelab.com/react-admin/) to provide a customizable administration interface.
+Here is their quick start example.
+The idea is to put the `Admin` component at the root level of the app.
+It is a provider that’s responsible for managing all data fetching and passing that data back to components.
+
+The second important component is `ListGuesser` which defines how the data should be displayed.
+
+```jsx
+import React from 'react';
+import { Admin, Resource, ListGuesser } from 'react-admin';
+import simpleRestProvider from 'ra-data-simple-rest';
+
+const dataProvider = simpleRestProvider('https://domain.tld/api');
+
+const App = () => (
+  <Admin dataProvider={dataProvider}>
+    <Resource name="users" list={ListGuesser} />
+  </Admin>
+);
+export default App;
+```
+
+If you’re unhappy with the rendering of the `ListGuesser`, then you can define your own components by reusing smaller components.
+If you’re unhappy with the smaller components, you can replace them with custom ones, and so on.
+
+So you start with a component that’s fully functional right out of the box, and you can rewrite any of its constituent elements as needed.
+
+This approach has **one major advantage**: it gives you a lot of flexibility.
+For example, you can easily modify the order of the components and their parent/children relationships.
+
+This approach has also **one major drawback**:it gives you a lot of flexibility.
+For example, you can easily modify the order of components in a bad way.
+_The more freedom, the more bugs_.
+
+##### Drawback example
+
+To show you how easy it is to make a mistake using this technique, here is a personal example involving Material UI components.
+
+I recently tried to wrap a `TextField` component in a `FormControl`, and was frustrated when it didn’t work.
+But the reason why is quite simple: the `TextField` component is itself composed of an input wrapped inside of a `FormControl`, and neither TypeScript nor `console.error` messages could warn me that my rendered markup was redundant and broken.
+
+```jsx
+<FormControl>
+  <TextField>
+</FormControl>
```

Comment:
TextField already includes FormControl

```suggestion
<TextField>
```

Reply:
my bad, rereading the content, I got it backward, to revert.

Reply:
I reverted it. Is it just fast reading or something is miss leading you?

Reply:
fast reading